qa/tasks/cephfs/xfstests_dev.py

```diff
@@ -124,47 +151,122 @@ def install_deps(self):
-            #export SCRATCH_MNT={}
-            export TEST_FS_MOUNT_OPTS="-o name=admin,secret={}"
-            ''').format(self.test_dev, self.test_dirs_mount_path, self.scratch_dev,
-                        self.scratch_dirs_mount_path, self.get_admin_key())
-
-        self.mount_a.client_remote.write_file(join(self.repo_path, 'local.config'),
+            export TEST_DEV={self.test_dev}
+            export TEST_DIR={self.test_dirs_mount_path}
+            export TEST_FS_MOUNT_OPTS="-o name=admin,secret={self.get_admin_key()}"
+            export SCRATCH_DEV={self.scratch_dev}
+            export SCRATCH_MNT={self.scratch_dirs_mount_path}
+            export MOUNT_OPTIONS=$TEST_FS_MOUNT_OPTS
+            ''')
+
+        config_path = join(self.repo_path, 'local.config')
+        self.mount_a.client_remote.write_file(config_path,
                                               xfstests_config_contents, sudo=True)
 
-    def tearDown(self):
-        self.mount_a.client_remote.run(args=['sudo', 'userdel', '--force',
-                                             '--remove', 'fsgqa'],
-                                       omit_sudo=False, check_status=False)
-        self.mount_a.client_remote.run(args=['sudo', 'userdel', '--force',
-                                             '--remove', '123456-fsgqa'],
-                                       omit_sudo=False, check_status=False)
-        self.mount_a.client_remote.run(args=['sudo', 'groupdel', 'fsgqa'],
-                                       omit_sudo=False, check_status=False)
+        # deliberately avoiding variable xfstests_config_contents to log
+        # actual, and not supposed, data in local.config.
+        contents = self.mount_a.client_remote.read_file(config_path).decode()
+        log.info(f'local.config\'s contents -\n{contents}')
 
-        self.mount_a.client_remote.run(args=['sudo', 'rm', '-rf',
-                                             self.repo_path],
-                                       omit_sudo=False, check_status=False)
+    #
+    # Following are helper methods that launch individual and groups of tests
+    # from xfstests-dev repo that is ready.
+    #
 
-        super(XFSTestsDev, self).tearDown()
+    # generic helper methods
+
+    def run_test(self, cmdargs, exit_on_err=False):
+        """
+        1. exit_on_err is same as check_status in terms of functionality, a
+           different name is used to prevent confusion.
+        2. exit_on_err is set to False to make sure all tests run whether or
+           not all tests pass.
+        """
+        cmd = 'sudo ./check ' + cmdargs
+        # XXX: some tests can take pretty long (more than 180 or 300 seconds),
+        # let's be explicit about timeout to save troubles later.
+        timeout = None
+        p = self.mount_a.run_shell(args=cmd, cwd=self.repo_path,
+            stdout=StringIO(), stderr=StringIO(), check_status=False,
+            omit_sudo=False, timeout=timeout)
```

Comment:
True but we are removing `sudo` from parameters of `run_shell()` on a different PR (since presence of both `omit_sudo` and `sudo` in parameters can be conflicting).

Reply:
OK. The dependencies b/w PRs is kind of making it hard for review :)

I understand the context now +1